assets/src/date/getTimeSensitiveDisplayDate.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { format } from '@wordpress/date';
+
+/**
+ * External dependencies
+ */
+import moment from 'moment-timezone';
+
+/**
+ * Internal dependencies
+ */
+import {
+  DEFAULT_DATE_FORMATTING,
+  getDateFormattedFromWordPressToMoment,
+  getTimeFromNow,
+  isToday,
+  isYesterday,
+} from './';
+
+// Returns hours since date if date passed in matches today
+// Returns yesterday if date passed in is yesterday
+// Returns formatted date + time by default
+/**
+ * @summary                        Formats a date to display relative to time passed since date.
```

Comment:
```suggestion
 * Formats a date to display relative to time passed since date.
```

We don't use `@summary` anywhere, but if we want to, it should be a concise summary of the full description.

I would consider the whole paragraph here to be the full description :-)

Maybe time to talk about JSDoc standards/guidelines for the project in general?